src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -1121,10 +1143,111 @@ TEST_CASE("custom_json_data: handling of optional tensors")
   tensors.push_back(model_tensor2);
 
   trrm.TestCustomData(
-      tensors, json_str, expected_results, expect_init_failure,
+      num_requests, tensors, json_str, expected_results, expect_init_failure,
       expect_thread_failure);
 }
 
+TEST_CASE("custom_json_data: multiple streams")
+{
+  PerfAnalyzerParameters params{};
+  params.user_data = {"fake_file.json"};
+  params.num_of_sequences = 1;
+  bool is_sequence_model{false};
+
+  std::vector<std::vector<int32_t>> expected_results;
+  std::vector<ModelTensor> tensors;
+  bool expect_init_failure = false;
+  bool expect_thread_failure = false;
+
+  ModelTensor model_tensor1{};
+  model_tensor1.datatype_ = "INT32";
+  model_tensor1.is_optional_ = false;
+  model_tensor1.is_shape_tensor_ = false;
+  model_tensor1.name_ = "INPUT1";
+  model_tensor1.shape_ = {1};
+
+  ModelTensor model_tensor2 = model_tensor1;
+  model_tensor2.name_ = "INPUT2";
+
+  std::string json_str{R"({
+  "data": [[
+    { "INPUT1": [1], "INPUT2": [21] },
+    { "INPUT1": [2], "INPUT2": [22] },
+    { "INPUT1": [3], "INPUT2": [23] }
+  ],[
+    { "INPUT1": [201], "INPUT2": [221] },
+    { "INPUT1": [202], "INPUT2": [222] }
+  ]]})"};
+
+  size_t num_requests = 10;
+
+  const auto& ParameterizeMemory{[&]() {
+    SUBCASE("No shared memory")
+    {
+      params.shared_memory_type = NO_SHARED_MEMORY;
+    }
+    SUBCASE("system shared memory")
+    {
+      params.shared_memory_type = SYSTEM_SHARED_MEMORY;
+    }
+    SUBCASE("cuda shared memory")
+    {
+      params.shared_memory_type = CUDA_SHARED_MEMORY;
+    }
+  }};
+
+  SUBCASE("yes sequence")
+  {
+    // Sequences will randomly pick among all streams
+    // (Although this test is hardcoded to pick ID 1 twice, and then ID 0
+    // forever after)
+    is_sequence_model = true;
+    expected_results = {{201, 221}, {202, 222}, {201, 221}, {202, 222},
+                        {1, 21},    {2, 22},    {3, 23},    {1, 21},
+                        {2, 22},    {3, 23}};
+    ParameterizeMemory();
+  }
+  SUBCASE("no sequence")
+  {
+    // For the case of no sequences, only a single data stream is supported. The
+    // rest will be ignored
+    is_sequence_model = false;
+    expected_results = {{1, 21}, {2, 22}, {3, 23}, {1, 21}, {2, 22},
+                        {3, 23}, {1, 21}, {2, 22}, {3, 23}, {1, 21}};
+    ParameterizeMemory();
+  }
+
+  TestRequestRateManager trrm(params, is_sequence_model);
+
+  tensors.push_back(model_tensor1);
+  tensors.push_back(model_tensor2);
+
+  trrm.CustomDataTestSetup(tensors, json_str, expect_init_failure);
+
+  if (!expect_init_failure) {
```

Comment:
Why is this value checked? It looks like it's hardcoded to false at the top and doesn't change anywhere.